modules/signals/src/signal-store.ts

```diff
@@ -325,16 +325,14 @@ export function signalStore(
         (this as any)[key] = props[key];
       }
 
-      if (hooks.onInit) {
-        hooks.onInit();
-      }
+      const { onInit, onDestroy } = hooks;
 
-      if (hooks.onDestroy) {
-        const injector = inject(Injector);
+      if (onInit) {
+        onInit();
+      }
 
-        inject(DestroyRef).onDestroy(() => {
-          runInInjectionContext(injector, hooks.onDestroy!);
-        });
+      if (onDestroy) {
+        inject(DestroyRef).onDestroy(onDestroy);
```

Comment:
Can we add the test(s) that verifies the fixed bug?

Reply:
Done

Reply:
I was thinking about verifying that the test will not throw an error when the store is provided in root and the `onDestroy` hook is used. Sorry if it wasn't clear!

Reply:
No problem, I've replaced the test. I hope it is better now.

Reply:
Thanks Rainer!